packages/contracts-bedrock/scripts/ChainAssertions.sol

```diff
@@ -121,6 +122,75 @@ library ChainAssertions {
         }
     }
 
+    /// @notice Asserts that the SystemConfigInterop is setup correctly
+    function checkSystemConfigInterop(
+        Types.ContractSet memory _contracts,
+        DeployConfig _cfg,
+        bool _isProxy
+    )
+        internal
+        view
+    {
+        console.log("Running chain assertions on the SystemConfigInterop");
+        SystemConfigInterop config = SystemConfigInterop(_contracts.SystemConfigInterop);
+
+        // Check that the contract is initialized
+        assertSlotValueIsOne({ _contractAddress: address(config), _slot: 0, _offset: 0 });
+
+        ResourceMetering.ResourceConfig memory resourceConfig = config.resourceConfig();
+
+        if (_isProxy) {
+            require(config.owner() == _cfg.finalSystemOwner());
+            require(config.overhead() == _cfg.gasPriceOracleOverhead());
+            require(config.scalar() == _cfg.gasPriceOracleScalar());
+            require(config.batcherHash() == bytes32(uint256(uint160(_cfg.batchSenderAddress()))));
+            require(config.gasLimit() == uint64(_cfg.l2GenesisBlockGasLimit()));
+            require(config.unsafeBlockSigner() == _cfg.p2pSequencerAddress());
+            // Check _config
+            ResourceMetering.ResourceConfig memory rconfig = Constants.DEFAULT_RESOURCE_CONFIG();
+            require(resourceConfig.maxResourceLimit == rconfig.maxResourceLimit);
+            require(resourceConfig.elasticityMultiplier == rconfig.elasticityMultiplier);
+            require(resourceConfig.baseFeeMaxChangeDenominator == rconfig.baseFeeMaxChangeDenominator);
+            require(resourceConfig.systemTxMaxGas == rconfig.systemTxMaxGas);
```

Comment:
Same feedback as portal PR, we don't need to introduce specific stuff, this is a copy/paste, this adds too much code. Its like a 5 loc change inside of `deploySystemConfig` that optionally deploys the interop version if its enabled